sdk/common/build.gradle.kts

```diff
@@ -30,15 +30,19 @@ dependencies {
   testImplementation("com.google.guava:guava-testlib")
 }
 
+val genResourceDir = "$buildDir/generated/resources/"
+
 sourceSets {
   main {
-    output.dir("build/generated/properties", "builtBy" to "generateVersionResource")
+    output.dir(genResourceDir, "builtBy" to "generateVersionResource")
   }
 }
 
+
 tasks {
   register("generateVersionResource") {
-    val propertiesDir = file("build/generated/properties/io/opentelemetry/sdk/common")
+    inputs.property("project.version", project.version)
+    val propertiesDir = file("$genResourceDir/io/opentelemetry/sdk")
```

Comment:
Note that this does not generate to `sdk/common` but `sdk/` which is where the Resource class (itself inside `common`) reads the file from.

Reply:
Was this broken, then?

Reply:
@Oberon00 Our intention is to have a properties file in each jar we publish, this can be used by spring boot or other tooling to e.g., verify dependencies are aligned correctly at runtime. Instead of coalescing these three artifacts to one file, we should just fix `Resource` to read from `/common` I guess. I think you're trying to avoid the ambiguity of different version numbers among the SDK artifacts, but that would only happen if there was a problem anyways with alignment in which case I think the reported version number is essentially undefined anyways.

`inputs.property` is a great change though, is it ok if I incorporate it into a change where I move all of these to a central location which I've meant to do for a while?